samples/snippets/src/main/java/com/example/analytics/RunReportWithPaginationSample.java

```diff
@@ -0,0 +1,105 @@
+    String propertyId = "YOUR-GA4-PROPERTY-ID";
+    sampleRunReportWithPagination(propertyId);
+  }
+
+  // Runs a report several times, each time retrieving a portion of result using pagination.
+  static void sampleRunReportWithPagination(String propertyId) throws Exception {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (BetaAnalyticsDataClient analyticsData = BetaAnalyticsDataClient.create()) {
+      // [START analyticsdata_run_report_with_pagination_page1]
+      RunReportRequest request =
+          RunReportRequest.newBuilder()
+              .setProperty("properties/" + propertyId)
+              .addDateRanges(DateRange.newBuilder().setStartDate("365daysAgo")
+                  .setEndDate("yesterday"))
+              .addDimensions(Dimension.newBuilder().setName("firstUserSource"))
+              .addDimensions(Dimension.newBuilder().setName("firstUserMedium"))
+              .addDimensions(Dimension.newBuilder().setName("firstUserCampaignName"))
+              .addMetrics(Metric.newBuilder().setName("sessions"))
+              .addMetrics(Metric.newBuilder().setName("conversions"))
+              .addMetrics(Metric.newBuilder().setName("totalRevenue"))
+              .setLimit(100000)
+              .setOffset(0)
+              .build();
+
+      // Make the request.
+      RunReportResponse response = analyticsData.runReport(request);
+      RunReportSample.printRunResponseResponse(response);
+      // [END analyticsdata_run_report_with_pagination_page1]
+
+      // Run the same report with a different offset value to retrieve the second page of a
+      // response.
+      // [START analyticsdata_run_report_with_pagination_page2]
+      request =
+          RunReportRequest.newBuilder()
+              .setProperty("properties/" + propertyId)
+              .addDateRanges(DateRange.newBuilder().setStartDate("365daysAgo")
+                  .setEndDate("yesterday"))
+              .addDimensions(Dimension.newBuilder().setName("firstUserSource"))
+              .addDimensions(Dimension.newBuilder().setName("firstUserMedium"))
+              .addDimensions(Dimension.newBuilder().setName("firstUserCampaignName"))
+              .addMetrics(Metric.newBuilder().setName("sessions"))
+              .addMetrics(Metric.newBuilder().setName("conversions"))
+              .addMetrics(Metric.newBuilder().setName("totalRevenue"))
+              .setLimit(100000)
+              .setOffset(100000)
+              .build();
+
+      // Make the request.
+      response = analyticsData.runReport(request);
+      // Prints the response using a method in RunReportSample.java
+      RunReportSample.printRunResponseResponse(response);
```

Comment:
Please remove the reference to the function outside of the snippet - snippets should aim to be "self contained": go/code-snippets-style#self-contained-copy-paste-runnable

If it's a bit of boilerplate that you think is detracting it from the sample, it's okay to have it in a separate function as long as it's present in each class - if it's not in the snippet most users won't see it.